axelrod/tests/strategies/test_titfortat.py

```diff
@@ -563,3 +563,52 @@ def test_strategy(self):
         actions = [(C, C), (C, C), (C, D), (C, D), (D, C), (D, D), (D, D),
                     (D, C), (D, C), (C, D), (C, D)]
         self.versus_test(opponent, expected_actions=actions)
+
+class TestAlexei(TestPlayer):
+    """
+    Tests for the Alexei strategy
+    """
+
+    name = "Alexei: ('D',)"
+    player = axelrod.Alexei
+    expected_classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': {'length'},
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_strategy(self):
+        self.first_play_test(C)
+        self.second_play_test(rCC=C, rCD=D, rDC=C, rDD=D)
+
+        actions = [(C, C), (C, D), (D, C), (C, D), (D, C)]
+        self.versus_test(axelrod.Alternator(), expected_actions=actions)
+
+        actions = [(C, C), (C, C), (C, C), (C, C), (D, C)]
+        self.versus_test(axelrod.Cooperator(), expected_actions=actions)
+
+        actions = [(C, D), (D, D), (D, D), (D, D), (D, D)]
+        self.versus_test(axelrod.Defector(), expected_actions=actions)
+
+        actions = [(C, C), (C, D), (D, C), (C, D), (D, C)]
+        self.versus_test(axelrod.Alternator(), expected_actions=actions,
+                         match_attributes={"length": -1})
+
+        actions = [(C, D), (D, D), (D, C), (C, C), (D, D)]
```

Comment:
I think all the test case will be covered by the cases above, we can simplify this and remove everything below.